<robot name="simple_robot">
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.01" radius="0.2"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="generic">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <link name="link1">
    <visual>
      <geometry>
        <cylinder length="0.4" radius="0.04"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <material name="generic_2">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint2" type="prismatic">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0 0 0.4"/>
    <axis xyz="1 0 0"/>
    <limit effort="300" velocity="0.1" lower="-0.2" upper="0"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <link name="link2">
    <visual>
      <geometry>
        <box size="0.25 0.03 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.125 0 0"/>
      <material name="generic_5">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>
<joint name="joint3" type="prismatic">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0.25 0 0"/>
    <axis xyz="0 0 -1"/>
    <limit effort="300" velocity="0.1" lower="0" upper="0.2"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <link name="link3">
    <visual>
      <geometry>
        <box size="0.25 0.03 0.03"/>
      </geometry>
      <origin rpy="0 1.57079 0" xyz="0 0 0.125"/>
      <material name="generic_5">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>
<joint name="joint4" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <origin xyz="0 0 0.25"/>
    <axis xyz="0 -1 0"/>
    <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <link name="link4">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.04"/>
      </geometry>
      <origin rpy="0 1.57079 0" xyz="0.15 0 0"/>
      <material name="generic_2">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
  </link>





<joint name="joint5" type="revolute">
    <parent link="link4"/>
    <child link="link5"/>
    <origin xyz="0.3 0 0"/>
    <axis xyz="0 -1 0"/>
    <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <link name="link5">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.04"/>
      </geometry>
      <origin rpy="0 1.57079 0" xyz="0.15 0 0"/>
      <material name="generic_2">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
  </link>



</robot>
